packages/@aws-cdk/aws-stepfunctions-tasks/lib/lambda/invoke.ts

```diff
@@ -100,16 +109,30 @@ export class LambdaInvoke extends sfn.TaskStateBase {
    * @internal
    */
   protected _renderTask(): any {
-    return {
-      Resource: integrationResourceArn('lambda', 'invoke', this.integrationPattern),
-      Parameters: sfn.FieldUtils.renderObject({
-        FunctionName: this.props.lambdaFunction.functionArn,
-        Payload: this.props.payload ? this.props.payload.value : sfn.TaskInput.fromDataAt('$').value,
-        InvocationType: this.props.invocationType,
-        ClientContext: this.props.clientContext,
-        Qualifier: this.props.qualifier,
-      }),
-    };
+    switch (this.props.resourceType) {
+      case 'Arn':
+        return {
+          Resource: this.props.lambdaFunction.functionArn,
+          Parameters: sfn.FieldUtils.renderObject({
+            Payload: this.props.payload ? this.props.payload.value : sfn.TaskInput.fromDataAt('$').value,
+            InvocationType: this.props.invocationType,
+            ClientContext: this.props.clientContext,
+            Qualifier: this.props.qualifier,
+          }),
+        };
+      case 'Invoke':
+      default:
+        return {
+          Resource: integrationResourceArn('lambda', 'invoke', this.integrationPattern),
```

Comment:
I think this will be a lot cleaner if we include a flag to fall back to the legacy behaviour of using the ARN